src/main/java/uk/gov/hmcts/reform/iahomeofficeintegrationapi/domain/handlers/presubmit/AsylumCaseNotificationHandler.java

```diff
@@ -0,0 +1,67 @@
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.iahomeofficeintegrationapi.domain.entities.AsylumCase;
+import uk.gov.hmcts.reform.iahomeofficeintegrationapi.domain.entities.MessageHeader;
+import uk.gov.hmcts.reform.iahomeofficeintegrationapi.domain.entities.ccd.Event;
+import uk.gov.hmcts.reform.iahomeofficeintegrationapi.domain.entities.ccd.callback.Callback;
+import uk.gov.hmcts.reform.iahomeofficeintegrationapi.domain.entities.ccd.callback.PreSubmitCallbackResponse;
+import uk.gov.hmcts.reform.iahomeofficeintegrationapi.domain.entities.ccd.callback.PreSubmitCallbackStage;
+import uk.gov.hmcts.reform.iahomeofficeintegrationapi.domain.handlers.PreSubmitCallbackHandler;
+import uk.gov.hmcts.reform.iahomeofficeintegrationapi.domain.service.HomeOfficeInstructService;
+
+@Component
+public class AsylumCaseNotificationHandler implements PreSubmitCallbackHandler<AsylumCase> {
+
+    private HomeOfficeInstructService homeOfficeInstructService;
+
+    public AsylumCaseNotificationHandler(HomeOfficeInstructService homeOfficeInstructService) {
+        this.homeOfficeInstructService = homeOfficeInstructService;
+    }
+
+    @Override
+    public boolean canHandle(PreSubmitCallbackStage callbackStage, Callback<AsylumCase> callback) {
+        requireNonNull(callbackStage, "callbackStage must not be null");
+        requireNonNull(callback, "callback must not be null");
+
+        return callbackStage == PreSubmitCallbackStage.ABOUT_TO_SUBMIT
+            && callback.getEvent() == Event.SUBMIT_APPEAL
+            && callback.getCaseDetails().getState() == APPEAL_SUBMITTED;
+    }
+
+    @Override
+    public PreSubmitCallbackResponse<AsylumCase> handle(
+        PreSubmitCallbackStage callbackStage, Callback<AsylumCase> callback) {
+
+        if (!canHandle(callbackStage, callback)) {
+            throw new IllegalStateException("Cannot handle callback");
+        }
+
+        AsylumCase asylumCase = callback.getCaseDetails().getCaseData();
+
+        final String homeOfficeReferenceNumber = asylumCase.read(HOME_OFFICE_REFERENCE_NUMBER, String.class)
+            .orElseThrow(() -> new IllegalStateException("Home office reference for the appeal is not present"));
+
+        final String caseId = asylumCase.read(APPEAL_REFERENCE_NUMBER, String.class)
+            .orElseThrow(() -> new IllegalStateException("Case ID for the appeal is not present"));
+
+        MessageHeader messageHeader = homeOfficeInstructService.sendNotification(homeOfficeReferenceNumber,caseId);
+        PreSubmitCallbackResponse<AsylumCase> response = new PreSubmitCallbackResponse<>(asylumCase);
+        if (messageHeader == null) {
+            asylumCase.write(HOME_OFFICE_INSTRUCT_STATUS,"Error sending notification to Home Office");
+            response.addErrors(Collections.singleton("Error sending notification to Home Office"));
+        } else {
+            asylumCase.write(HOME_OFFICE_INSTRUCT_STATUS,"Notification sent to Home Office");
```

Comment:
This can probably be improved.  What are we trying to store in the case data here?  That a notification was successfully sent?  Do we need a timestamp or any other info or would an enum with a string Status (e.g. SUCCESS/FAIL) be sufficient?

Reply:
We do not get any timestamp details from home office. so, status 200 OK is all what we have. I will change it to boolean

Reply:
I don't think a boolean is appropriate.  The user will need to know the overall status of a status request to the Home office but also will want to know if it is retry-able or not.  For example if we get a 404 or something indicating that the UAN we pass is invalid then this is useful for the user to see and not something we will want to retry.  
I think for the time being we can put the http status code or string here e.g. 200/404/500 or OK/Not Found/Internal Server Error and then we can decide how we use that in CCD.  E.g. show an appropriate message based on this.

Reply:
Saving the status as OK/Not Found/Internal Server Error